memstore/memstore.go

```diff
@@ -281,6 +281,17 @@ func (m *memStoreImpl) AddTableShard(table string, shardID int, needPeerCopy boo
 		if needPeerCopy {
 			tableShard.needPeerCopy = 1
 		}
+
+		// purge to make sure disk space is clean for new table shard when it is added
+		if needPurge {
+			if err := tableShard.diskStore.DeleteTableShard(table, shardID); err != nil {
+				utils.GetLogger().With("table", table, "shard", shardID, "error", err.Error()).Fatalf("failed to purge table shard data")
```

Comment:
is logging enough?
maybe emit metrics so we can create alerts on it.

Reply:
this will be fatal and crash the server

Reply:
oops, didn't realize it's Fatal..